ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/layered/BaseFeePrioritizedTransactions.java

```diff
@@ -146,11 +149,34 @@ private Wei calculateNextBlockBaseFee(final FeeMarket feeMarket, final BlockHead
 
   @Override
   protected boolean promotionFilter(final PendingTransaction pendingTransaction) {
-    return nextBlockBaseFee
-        .map(
-            baseFee ->
-                pendingTransaction.getTransaction().getMaxGasPrice().greaterOrEqualThan(baseFee))
-        .orElse(false);
+    // check if the tx is willing to pay at least the base fee
+    if (nextBlockBaseFee
+        .map(pendingTransaction.getTransaction().getMaxGasPrice()::lessThan)
+        .orElse(true)) {
+      return false;
+    }
+
+    // priority txs are promoted even if they pay less
+    if (!pendingTransaction.hasPriority()) {
+      // check if max fee per gas is higher that the min gas price
+      if (pendingTransaction
+          .getTransaction()
+          .getMaxGasPrice()
+          .lessThan(miningParameters.getMinTransactionGasPrice())) {
+        return false;
+      }
+
+      // check if enough priority fee is paid
+      if (!miningParameters.getMinPriorityFeePerGas().equals(Wei.ZERO)) {
```

Comment:
Why might we promote the tx when the min priority fee is 0?

Reply:
this is just an optimization, since by default `minPriorityFeePerGas` is 0, then if the default is not changed we can just skip the check